pkg/tracing/tracing.go

```diff
@@ -10,7 +10,9 @@ import (
 )
 
 // ForceTracingBaggageKey - force sampling header.
-const ForceTracingBaggageKey = "X-Thanos-Force-Tracing"
+// Lower kebab-case for Jaeger support, per documentation:
+// https://www.jaegertracing.io/docs/1.22/client-libraries/#baggage
+const ForceTracingBaggageKey = "x-thanos-force-tracing"
```

Comment:
I wonder.. can we maybe support both? It will not introduce breaking change for user too. Just let's lower case the input before checking?

Reply:
I considered that, but I wasn't sure I saw the benefit - now I think further, backwards compatibility is preferable, for sure.

I'm not 100% sure I'm envisioning it right, so I'll just check:

```golang
pkg/tracing/http.go
		// If client specified ForceTracingBaggageKey header, ensure span includes it to force tracing.
		span.SetBaggageItem(ForceTracingBaggageKey, r.Header.Get(ForceTracingBaggageKey))
```

There we'd set to `span.SetBaggageItem(strings.ToLower(ForceTracingBaggageKey),...`

Then set the` JaegerDebugHeader` in to `ToLower` as well.

The only other place I can see that is a `forceRecorder` which itself is used for a basic tracer wrapping a gcloud tracer (stackdriver), so I don't see it necessary there?